File: php/phpfind/src/phpfind/FindConfig.php
<?php

declare(strict_types=1);

namespace phpfind;

/**
 * Class FindConfig
 */
readonly class FindConfig
{
    public string $file_types_path;
    public string $find_options_path;
    public string $default_find_settings_path;

    public function __construct()
    {
        $home = getenv('HOME');
        $xfind_config_dir = getenv('XFIND_CONFIG_DIR');
        if (!$xfind_config_dir) {
            $xfind_config_dir = FileUtil::join_paths($home, '.config', 'xfind');
        }

        $resources_path = FileUtil::join_paths(__DIR__, '..', '..', 'resources');
        $file_types_path = FileUtil::join_paths($resources_path, 'filetypes.json');
        $find_options_path = FileUtil::join_paths($resources_path, 'findoptions.json');
        $default_find_settings_path = FileUtil::join_paths($xfind_config_dir, 'settings.json');

        $this->file_types_path = $file_types_path;
        $this->find_options_path = $find_options_path;
        $this->default_find_settings_path = $default_find_settings_path;
    }
}

File: python/pyfind/pyfind/findconfig.py
# -*- coding: utf-8 -*-
"""
###############################################################################
#
# findconfig.py
#
# Configuration values
#
###############################################################################
"""
import importlib.resources
import os


class FindConfig:
    """FindConfig holds basic configuration."""

    __slots__ = ['file_types_path', 'find_options_path', 'default_find_settings_path']

    def __init__(self):
        """Create a new FindConfig instance."""

        # pyfind data package resources
        _data = importlib.resources.files('pyfind').joinpath('data')
        _file_types_path = _data.joinpath('filetypes.json')
        _find_options_path = _data.joinpath('findoptions.json')

        _home = os.getenv('HOME', '')
        _default_find_config_dir = os.path.join(_home, '.config', 'xfind')
        _xfind_config_dir = os.getenv('XFIND_CONFIG_DIR', _default_find_config_dir)
        _default_settings_path = os.path.join(_xfind_config_dir, 'settings.json')

        self.file_types_path = _file_types_path
        self.find_options_path = _find_options_path
        self.default_find_settings_path = _default_settings_path
